pkg/devfile/adapters/docker/component/utils.go

```diff
@@ -25,36 +25,17 @@ import (
 )
 
 const (
-	localhostIP             = "127.0.0.1"
-	projectSourceVolumeName = "odo-project-source"
+	localhostIP = "127.0.0.1"
 )
 
 func (a Adapter) createComponent() (err error) {
 	componentName := a.ComponentName
 
 	log.Infof("\nCreating Docker resources for component %s", a.ComponentName)
 
-	// Get or create the project source volume
-	var projectVolumeName string
-	projectVolumeLabels := utils.GetProjectVolumeLabels(componentName)
-	projectVols, err := a.Client.GetVolumesByLabel(projectVolumeLabels)
+	projectVolumeName, err := utils.CreateAndGetProjectVolume(a.Client, componentName)
 	if err != nil {
-		return errors.Wrapf(err, "Unable to retrieve source volume for component "+componentName)
-	}
-	if len(projectVols) == 0 {
-		// A source volume needs to be created
-		projectVolumeName, err = storage.GenerateVolName(projectSourceVolumeName, a.ComponentName)
-		if err != nil {
-			return errors.Wrapf(err, "Unable to generate project source volume name for component %s", componentName)
-		}
-		_, err := a.Client.CreateVolume(projectVolumeName, projectVolumeLabels)
-		if err != nil {
-			return errors.Wrapf(err, "Unable to create project source volume for component %s", componentName)
-		}
-	} else if len(projectVols) == 1 {
-		projectVolumeName = projectVols[0].Name
-	} else if len(projectVols) > 1 {
-		return errors.Wrapf(err, "Error, multiple source volumes found for component %s", componentName)
+		return errors.Wrapf(err, "Unable to determine the project source volume for component %s", componentName)
```

Comment:
We shouldn't capitalize the beginning of error messages

Reply:
done